app/controllers/users_controller.rb

```diff
@@ -5,6 +5,23 @@ def index
     user_addresses
   end
 
+  def show
+    find_user
+  end
+
+  def edit
+    find_user
+  end
+
+  def update
+    find_user
+    if @user.update(user_params)
+      redirect_to @user
+    else
+      redirecto_to 'edit'
```

Comment:
you have to use render instead of redirect, so you don't lose the changes, try both and see the differences